[STRAT-100] Completed generation of heartland elevations

krishna-kalavadia opened the pull request:
In Depth description of point compression algorithm can be found at https://docs.microsoft.com/en-us/bingmaps/rest-services/elevations/point-compression-algorithm

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+import sys
+import os.path
+from csv import reader
+from config import API_KEY
```
Maybe try this to bypass the flake8 error.

```suggestion
# from config import API_KEY
API_KEY = ""
```

e-wai replied:
Never mind, it seems ok with it

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+# This just generates a dict object for the coordinates to resolve some type errors
+COORDINATES_LISTX = []
+COORDINATES_LISTY = []
+COORDINATES_LIST = []
```
new line here please

krishna-kalavadia replied:
I moved this section under if __init__== "__main__":

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+
+
+class ElevationDataRetrieval:
+    def __init__(self, wps: list, ht='sealevel'):
```
Type anotation says list but docstring says wps is a dictionary?

ufshk replied:
this is technically my mistake, it should've been a dict from my file but I somehow put it in as a list, they missed correcting that

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+        """
+        Initialize MapData
+        @param wps: dictionary of major waypoints inputted by user
+        @param ht: elevations response height model option
```
Can you add quick information on what the options are?

krishna-kalavadia replied:
Hey I apologize about this I did not add more information because I didn't want to say the wrong thing I don't know much in-depth info about this module, I would need to talk about Umar about it, it'll be dealt with in the refactor hopefully?

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+        # write dataframe into csv and return true
+        elevation_df.to_csv(ELEVATIONS_FILE)
+        print(f'Elevations written to CSV file.')
+        return True
```
A lot of this is repetitive from get_elevations.py. Why did you choose to create a close copy rather than inheriting?

krishna-kalavadia replied:
Again I apologize for this the inherited copy would require me to make some changes to original get_elevations and the changes I would need to make broke some tests and I would need to redesign the tests. This just ensured I got resulting elevations we needed.

e-wai commented on routemodel/data_retrieval/get_elevations_point_compression.py:
```diff
@@ -0,0 +1,165 @@
+        print(f'Elevations written to CSV file.')
+        return True
+
+# method calls to generate the heartland elevations
```
maybe add an `if __init__== "__main__":`?

e-wai commented:
Could you also run `pytest && cd routemodel && pytest && cd tests && pytest` on your computer please?

krishna-kalavadia commented:
> Could you also run `pytest && cd routemodel && pytest && cd tests && pytest` on your computer please?

Sure!

krishna-kalavadia commented:
Inheritance did not go well and broke tests so here’s a list of changes we needed to make to get_elevations but are made in this standalone file:

1. Added point compression algorithm to deal with larger quantities of points 
2. Added a new modified_get_elevation_data. This was needed because since we added a point compression algorithm, we needed  to build the request URL differently than was previously done Also needed to modify the return statement of the original get_elevation_data to extract JSON data to resolve errors, could not tell you why this works but passing in the entire response into parse_elevations creates issues 
3. Modified elevations assignment in parse_elevation_data  since we already converted response to a dict
4. Added an index_check variables and a break statement in the loop of parse_elevation_data to deal with duplicate points. These duplicate points mean that the waypoints member of the object has more points than returned by the api and so there are less points in elevations 
5. Added break statement if minorway points is set to None in parse elevations since we did not use it and it was causing errors

e-wai reviewed (approved):
I think it looks good to me, but I'll tell Michael some time to take a look

mdshiozaki reviewed (approved):
Looks good to me as well, comments I had were noted by Emma and it's good, will merge.
